src/decoder/sampletransform.rs

```diff
@@ -50,7 +50,20 @@ impl SampleTransformBinaryOp {
             SampleTransformBinaryOp::Or => left | right,
             SampleTransformBinaryOp::Xor => left ^ right,
             SampleTransformBinaryOp::Pow => {
-                if left == 0 {
+                if left == 0 || left == 1 {
+                    left
+                } else if right == 0 {
+                    1
+                } else if right == 1 {
+                    left
+                } else if left == -1 {
+                    if right % 2 == 0 {
+                        1
+                    } else {
+                        -1
+                    }
+                } else if right < 0 {
+                    // L^R is in ]0:1[ here, so truncating it always gives 0.
```

Comment:
The interval should be the open interval from -1 to 1. For example, (-2)^(-1) = 1/(-2) = -0.5.